doc/RFC-MMP.llnl.04.txt

```diff
@@ -0,0 +1,913 @@
+=====================================================================
+2016-07-28 Draft 4		ZFS Multi-{Modifier,Mount} Protection
+=====================================================================
+Originally created 2009 by ricardo.correia@oracle.com
+  ** references 1
+Revised 2017 by {faaland1,bass6,behlendorf1}@llnl.gov
+=====================================================================
+
+As stated in RFE 6710718 (multi-import protection for the DMU), it would be
```

Comment:
I'm OK with adding this kind of documentation to the source tree.  Once finished we should add a link to the latest version of the document from the [OpenZFS Developer Resources](http://open-zfs.org/wiki/Developer_resources) page.  Writing the document with minimal markdown would also be a good idea.